addon/model.js

```diff
@@ -151,10 +148,38 @@ export default class MegamorphicModel extends EmberObject {
     return YesManAttributes;
   }
 
-  static eachRelationship(/* callback */) {}
+  /*
+  static create(properties) {
+    return new this(properties);
+  }
+  */
 
   get _modelName() {
-    return this._internalModel.modelName;
+    if (CUSTOM_MODEL_CLASS) {
+      return this._recordData.modelName;
+    } else {
+      return this._internalModel.modelName;
+    }
+  }
+
+  _updateCurrentState(state) {
+    if (CUSTOM_MODEL_CLASS) {
+      notifyPropertyChange(this, 'isDirty');
+      notifyPropertyChange(this, 'isDeleted');
+      notifyPropertyChange(this, 'isNew');
+      // TODO need to walk the chain down as well to notify changes
```

Comment:
Seems like this might not be needed, couldn't get a test to fail with current behavior. Will leave a todo in for now, and fully remove after we have tested it in apps